### PR TITLE
Add more Activator tests

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

File: src/System.Runtime/tests/System/ActivatorTests.cs [hunk 1]
```diff
@@ -64,41 +66,202 @@ public static void CreateInstance_ConstructorWithParamsParameter()
         }
 
         [Fact]
-        public static void CreateInstance_Invalid()
+        public void CreateInstance_ValueTypeWithPublicDefaultConstructor_Success()
         {
-            Type nullType = null;
-            AssertExtensions.Throws<ArgumentNullException>("type", () => Activator.CreateInstance(nullType)); // Type is null
-            AssertExtensions.Throws<ArgumentNullException>("type", () => Activator.CreateInstance(null, new object[0])); // Type is null
+            // Test caching behaviour by activating multiple times.
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/ActivatorTests.cs]: What caching behavior is this validating / testing?

File: src/System.Runtime/tests/System/ActivatorTests.cs [hunk 2]
```diff
@@ -64,41 +66,202 @@ public static void CreateInstance_ConstructorWithParamsParameter()
...
+            Assert.IsType<ValueTypeWithDefaultConstructor>(Activator.CreateInstance(typeof(ValueTypeWithDefaultConstructor)));
+            Assert.IsType<ValueTypeWithDefaultConstructor>(Activator.CreateInstance(typeof(ValueTypeWithDefaultConstructor), nonPublic: true));
+            Assert.IsType<ValueTypeWithDefaultConstructor>(Activator.CreateInstance(typeof(ValueTypeWithDefaultConstructor), nonPublic: false));
+        }
+
+        [Fact]
+        public void CreateInstance_NonPublicClassWithPrivateDefaultConstructor_Success()
+        {
+            // Test caching behaviour by activating multiple times.
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/ActivatorTests.cs]: Same question

File: src/System.Runtime/tests/System/ActivatorTests.cs [hunk 3]
```diff
@@ -64,41 +66,202 @@ public static void CreateInstance_ConstructorWithParamsParameter()
...
+        public static void CreateInstance_MultipleMatchingConstructors_ThrowsAmbiguousMatchException()
+        {
             Assert.Throws<AmbiguousMatchException>(() => Activator.CreateInstance(typeof(Choice1), new object[] { null }));
+        }
 
+#if netcoreapp || uapaot
+        [Fact]
+        public void CreateInstance_NotRuntimeType_ThrowsArgumentException()
+        {
+            foreach (Type nonRuntimeType in Helpers.NonRuntimeTypes)
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/ActivatorTests.cs]: Can this be a Theory instead?

File: src/System.Runtime/tests/System/ActivatorTests.cs [hunk 4]
```diff
@@ -64,41 +66,202 @@ public static void CreateInstance_ConstructorWithParamsParameter()
...
+            AssertExtensions.Throws<ArgumentException>(null, () => Activator.CreateInstance(type, new object[0]));
+        }
+
+        public static IEnumerable<object[]> CreateInstance_InvalidType_TestData()
+        {
+            yield return new object[] { typeof(void) };
+            yield return new object[] { typeof(void).MakeArrayType() };
+            yield return new object[] { Type.GetType("System.ArgIterator") };
+            // Fails with TypeLoadException in .NET Core
+            // yield return new object[] { Type.GetType("System.ArgIterator").MakeArrayType() };
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/ActivatorTests.cs]: We should check this in just commented out.  If there's an issue for it, we should include the issue # here.  If it's by design, we should delete these lines.

File: src/System.Runtime/tests/System/ActivatorTests.cs (hunk 1 above)
Comment (hughbe) [reply src/System.Runtime/tests/System/ActivatorTests.cs]: https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/src/System/RtType.cs#L4768-L4781 for example, but I clarified this comment

File: src/System.Runtime/tests/System/ActivatorTests.cs (hunk 3 above)
Comment (hughbe) [reply src/System.Runtime/tests/System/ActivatorTests.cs]: No, due to https://github.com/xunit/xunit/issues/1325

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
Thanks.

**Pull request merged**